Tests/Services/RouteAudioServiceTests.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  RouteAudioServiceTests.swift
+//  ModernAVPlayer_Tests
+//
+//  Created by raphael ankierman on 03/05/2018.
+//  Copyright © 2018 CocoaPods. All rights reserved.
+//
+
+import AVFoundation
+import Quick
+@testable
+import ModernAVPlayer
+import Nimble
+
+final class RouteAudioServiceTests: QuickSpec {
+    
+    var tested: RouteAudioService!
+    var routeChangedReason: AVAudioSessionRouteChangeReason?
+    
+    override func spec() {
+        
+        beforeEach {
+            self.routeChangedReason = nil
+            self.tested = RouteAudioService()
+            self.tested.onRouteChanged = { [weak self] in self?.routeChangedReason = $0 }
+        }
+        
+        describe("initialization") {
+            context("valid audio route changed reason") {
+                it("should register to AVAudioSessionRouteChange notification ") {
+                    
+                    // ARRANGE
+                    let info: [String: UInt] = [AVAudioSessionRouteChangeReasonKey: 2]
+                    var notif = Notification(name: NSNotification.Name.AVAudioSessionRouteChange)
+                    notif.userInfo = info
+                    
+                    // ACT
+                    NotificationCenter.default.post(notif)
+                    
+                    // ASSERT
+                    expect(self.routeChangedReason).to(equal(AVAudioSessionRouteChangeReason.oldDeviceUnavailable))
+                }
+            }
+            
+            context("invalid audio route changed reason") {
```

Comment:
To be exhaustive, you have 3 failing tests:
-> notification has no UserInfo
-> value of 'info' dictionary is not a Uint (set a negative number for example)
-> value is a Int not defined for AVAudioSessionRouteChangedReason

You have just test the third case here

Reply:
I add some tests.
the third is actually false. init with an UInt not define in AVAudioSessionRouteChangedReason still return an "AVAudioSessionRouteChangedReason"